grails-app/controllers/au/org/ala/biocache/hubs/OccurrenceController.groovy

```diff
@@ -141,9 +174,35 @@ class OccurrenceController {
                         remove = k
                     }
                 }
-                if (remove) searchResults?.activeFacetMap?.remove(remove)
+                if (remove) {
+                    searchResults?.activeFacetMap?.remove(remove)
+                    searchResults?.activeFacetObj?.remove(remove)
+                }
+            }
+
+            Map activeDefaultFilters = [:]
+            Map activeUserFilters = [:]
+            searchResults?.activeFacetObj.each { k, v ->
+                // v is Facet list here, grouped by filtername
+                // fina all user filters
+                def userFilters = v.findAll { !fqToCategoryMap.containsKey(it.displayName) }
+                if (!userFilters.isEmpty())  {
+                    activeUserFilters[k] = userFilters
+                }
+
+                v.findAll{ fqToCategoryMap.containsKey(it.displayName) }?.each { facet ->
+                    activeDefaultFilters.get(fqToCategoryMap[facet.displayName], []).add(facet)
+                }
             }
 
+            // front end gets list of facets from activeFacetMap
+            // doing this so that front end doesn't need code change
+            // it now includes only user filters
+            searchResults.activeFacetMap = new JSONObject(new Gson().toJson(activeUserFilters))
+            // applied default filters list
+            // its grouped by category like 'location' -> [fq, fq, fq]
+            def activeDefaultFiltersJson = new JSONObject(new Gson().toJson(activeDefaultFilters))
```

Comment:
Does this pattern remove any of the facets used by default filters from view in the interface, even when the user specifically chose the filter themselves?